docs/source/en/using-diffusers/loading.mdx

```diff
@@ -191,227 +132,189 @@ model_id = "runwayml/stable-diffusion-v1-5"
+</Tip>
 
-Pipeline checkpoints that are identical in model structure, but have been trained on different datasets, trained with vastly different training setups and thus correspond to different official releases (such as [Stable Diffusion v1-4](https://huggingface.co/CompVis/stable-diffusion-v1-4) and [Stable Diffusion v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5)) should probably be stored in individual repositories instead of as variations of each other.
+Otherwise, a variant is **identical** to the original checkpoint. They have exactly the same serialization format (like [Safetensors](./using-diffusers/using_safetensors)), model structure, and weights have identical tensor shapes.
 
-#### So what are checkpoint variants then? 
+| **checkpoint type** | **weight name**                     | **argument for loading weights** |
+|---------------------|-------------------------------------|----------------------------------|
+| original            | diffusion_pytorch_model.bin         |                                  |
+| floating point      | diffusion_pytorch_model.fp16.bin    | `variant`, `torch_dtype`         |
+| non-EMA             | diffusion_pytorch_model.non_ema.bin | `variant`                        |
 
-Checkpoint variants usually consist of the checkpoint stored in "*low-precision, low-storage*" dtype so that less bandwith is required to download them, or of *non-exponential-averaged* weights that shall be used when continuing fine-tuning from the checkpoint.
-Both use cases have clear advantages when their weights are considered variants: they share the same serialization format as the reference weights, and they correspond to a specialization of the "main" checkpoint which does not warrant a new model repository. 
-A checkpoint stored in [torch's half-precision / float16 format](https://pytorch.org/blog/accelerating-training-on-nvidia-gpus-with-pytorch-automatic-mixed-precision/) requires only half the bandwith and storage when downloading the checkpoint,
-**but** cannot be used when continuing training or when running the checkpoint on CPU.
-Similarly the *non-exponential-averaged* (or non-EMA) version of the checkpoint should be used when continuing fine-tuning of the model checkpoint, **but** should not be used when using the checkpoint for inference. 
+Load a variant by specifying the `variant` argument in [`DiffusionPipeline.from_pretrained`], and the `torch_dtype` if the variant is a different floating point type. 🧨 Diffusers won't download a variant unless it is explicitly specified, so you don't have to worry about downloading and caching more checkpoints than you need.
```

Comment:
Here we need to be more precise I think.

`variant` -> this just defines what files should be loaded
`torch_dtype` -> this defines into what `torch_dtype` the weights should be converted to. 

Can we maybe try to say something along the following:
```suggestion
Load a variant by specifying the `variant` argument in [`DiffusionPipeline.from_pretrained`]. 🧨 Diffusers won't download a variant unless it is explicitly specified, so you don't have to worry about unnecessarily downloading and caching more checkpoints than you need.

 The `torch_dtype` argument is unrelated to the `variant` argument and decides which floating point precision the loaded checkpoints should have. If you want to save bandwidth by loading a `"fp16"` variant, you should also define `torch_dtype=torch.float16` as otherwise the fp16 weights will be converted to the default fp32 precision. Note that you can also load the original checkpoint without setting a variant and then converting it to float16 precision by passing `torch_dtype=torch.float16`. In this case you downloaded float32 weights and converted them to float16 after loading.
```

Reply:
It would be great if the reader could understand the difference between `torch_dtype` and `variant` here (they are completely orthogonal.

Reply:
Thanks for the explanation, this is super helpful!